util/app/path/path.go

```diff
@@ -27,3 +28,56 @@ func Path(root, path string) (string, error) {
 	}
 	return appPath, nil
 }
+
+const outOfBoundsError string = "Out of bounds symlink found"
+
+// ContainsOutOfBoundsSymlinks determines if basePath contains any symlinks that
+// are absolute or point to a path outside of the basePath.
+func ContainsOutOfBoundsSymlinks(basePath string) (bool, string, error) {
```

Comment:
For the places that return an error, can you return `fmt.Errorf(...` to add a bit of context to the error message? Should make debugging easier.